### ISSUE TITLE
Html documentation of Fable API

### ISSUE THREAD

**Issue by codec-abc** [fable-compiler/Fable#731]
Hi,

It would useful for beginners to have a full HTML documentation hosted on the fable.io website. Being a beginner, I am currently learning by doing. But by not knowing the functions available in fable (and libraries such a fable-arch) and their types signatures I have to search the code which is time consuming.

Ideally, for Fable and related libraries it would be cool if we could have :
- The full API of Fable (modules, functions and types) on the website up to date with the latest release.
- Type annotation in code sample

I think having an auto-updated documentation should be quite easy using a CI hook. So the harder part is how to generate the documentation itself.

I tried for the whole day to build a prototype using [FSharp.Formatting](https://tpetricek.github.io/FSharp.Formatting/) which should solve the two points, but beside having the library running without error I did not manage to get a single web page except an empty one.

EDIT :  Trying even more I finally managed to generate documentation for Fable using FSharp.Formatting.

**Comment by alfonsogarciacaro** [comment]
Hi @codec-abc! Yes, having good documentation is very important. Fable is evolving very rapidly and it's difficult for me to keep everything up-to-date while adding many new features, fixing issues, providing support, etc, so any help is really appreciated! 😄 Right now I'm working on releasing 1.0 with the definitive changes after user feedback, so we can have a more stable version and I can focus on documentation and tutorials.

About the API, there're no many Fable-specific methods. Most of them can be found in [this file](https://github.com/fable-compiler/Fable/blob/d1f9b33eb56fa3527ff0410f1ebb35e83b064e26/src/fable/Fable.Core/Fable.Core.fs) and explained in [this page](http://fable.io/docs/interacting.html). For most other cases, Fable just provides replacements for the types and methods in FSharp.Core and .NET Base Class Library. You can see which types and methods are compatible [in this page](http://fable.io/docs/compatibility.html).

About the code samples, for [those posted on the web](http://fable.io/samples.html) you should see the type annotations by hovering the cursor on the functions.

**Comment by codec-abc** [comment]
You are correct, type annotation works on the fable website for the most part (some are broken thought). If that's ok for you I will try to come up with a solution that will generate the API documentation with Fsharp.Formatting as a build task since it is already in the paket dependencies. Then I will try to find a way to put those pages in the website automatically (I not yet fully understand how the website is generated and update).

**Comment by alfonsogarciacaro** [comment]
@codec-abc That'd be awesome! If you don't mind, please wait a bit as I want to release soon the beta for Fable 1.0 and this will introduce several changes in the internal code and the way to compile the projects. Then we can start simplifying the way the website is generated (including the REPL) to update the docs :+1:

One question: when you say Fable.API, do you mean the Fable.Core methods that can be used in JS projects? Or the methods to create Fable compiler plugins?

**Comment by codec-abc** [comment]
Thanks for the answer. For the Fable API I mean Fable.Core but if the tool is able to generate documentation for Fable compiler plugins, then why not. About the timing you are probably right, as such I will still run some tests to gain experience with the tooling but I will wait the release of 1.0 before doing anything serious.

**Comment by alfonsogarciacaro** [comment]
@codec-abc Sorry for the late response. If you still want to generate documentation for the API, could you please coordinate with the @fable-compiler/infrastructure and/or @fable-compiler/documentation teams to decide what's the best way to do that? Also we have #887 for the new website, so we can merge the two issues :+1:

**Comment by codec-abc** [comment]
Thanks for taking time to look back at my issue :) . If I remember correctly I was able (after a lot of fighting) to use FSharp.Formatting to generate the whole API documentation. The problem is that I don't really have experience with the F# tooling. To be honest, I don't use F# and Fable that much, not that I don't like them (quite the contrary to be honest) but because I don't have many opportunities to use it. As such, I don't feel confident enough to going back to work on this issue on my own, but if you want me to look back at using FSharp.Formatting to help someone getting started for this task I will gladly help. Other than that, I noticed that certain code sample have partial (or none at all) type annotations. For example [on this page](http://fable.io/docs/interacting.html), there is a lot of code samples that have partial of no type annotations. While in some cases it is obvious that no annotation can be given (typically for dynamic programming), there is some lines where I think some annotations can be added. For example, I think the`createObj` function can have type annotations while it has not.
